homeassistant/util/unit_system.py

```diff
@@ -183,13 +183,13 @@ def volume(self, volume: float | None, from_unit: str) -> float:
     def as_dict(self) -> dict[str, str]:
         """Convert the unit system to a dictionary."""
         return {
-            LENGTH: self.length_unit,
-            ACCUMULATED_PRECIPITATION: self.accumulated_precipitation_unit,
-            MASS: self.mass_unit,
-            PRESSURE: self.pressure_unit,
-            TEMPERATURE: self.temperature_unit,
-            VOLUME: self.volume_unit,
-            WIND_SPEED: self.wind_speed_unit,
+            LENGTH: self.length_unit.value,
```

Comment:
I'm not following this change, StrEnum should be fully compatible?

Reply:
The orjson serializer has to jump back out of the native code fallback to the the pure python converter https://github.com/home-assistant/core/blob/cb1f365482b1e8b69d42e30b84f8cfb28411d0c3/homeassistant/helpers/json.py#L43 for every one of these because it can't natively serialized it.  

Its only worth changing for places where we serialize the same thing over and over again. We can't cache Config so it always end up in the profile

Reply:
According to the orjson documentation it supports enums; the caveat is that it only supports enums with members with a supported type. In this case, those are strings.

So, why shouldn't we just make that working?

Reply:
Thats a good question.  I wonder if its not working right and falling back because of our StrEnum backport.

Reply:
```
import enum, orjson

from homeassistant.backports import enum as backports_enum


class NativeStrEnum(enum.StrEnum):
    NATIVE = "native"


class BackportStrEnum(backports_enum.StrEnum):
    BACKPORT = "backport"


print(orjson.dumps(NativeStrEnum.NATIVE))
print(orjson.dumps(BackportStrEnum.BACKPORT))
```

This works exactly as expected, but I still see Config calling the fallback code in the profile.

I'll do a deeper investigation and leave this as a draft until I figure out what the root cause is

Reply:
```diff
diff --git a/homeassistant/core.py b/homeassistant/core.py
index 54f44d0998..5b02f14bcd 100644
--- a/homeassistant/core.py
+++ b/homeassistant/core.py
@@ -2142,12 +2142,12 @@ class Config:
             "unit_system": self.units.as_dict(),
             "location_name": self.location_name,
             "time_zone": self.time_zone,
-            "components": self.components,
+            "components": list(self.components),
             "config_dir": self.config_dir,
             # legacy, backwards compat
-            "whitelist_external_dirs": self.allowlist_external_dirs,
-            "allowlist_external_dirs": self.allowlist_external_dirs,
-            "allowlist_external_urls": self.allowlist_external_urls,
+            "whitelist_external_dirs": list(self.allowlist_external_dirs),
+            "allowlist_external_dirs": list(self.allowlist_external_dirs),
+            "allowlist_external_urls": list(self.allowlist_external_urls),
             "version": __version__,
             "config_source": self.config_source,
             "safe_mode": self.safe_mode,
```
It doesn't do the fallback if I wrap the sets in list

Reply:
```python
import enum, orjson
import asyncio
from homeassistant.backports import enum as backports_enum
from homeassistant.core import Config, HomeAssistant


class NativeStrEnum(enum.StrEnum):
    NATIVE = "native"


class BackportStrEnum(backports_enum.StrEnum):
    BACKPORT = "backport"


print(orjson.dumps(NativeStrEnum.NATIVE))
print(orjson.dumps(BackportStrEnum.BACKPORT))


async def test():
    hass = HomeAssistant()

    my_config = Config(hass)

    xx = my_config.as_dict()
    print(xx)
    print(orjson.dumps(xx))


asyncio.run(test())
```

I reverted the enum changes.  The above now runs without any fallback

Reply:
```
supervisor/api/proxy.py:        if not await self.sys_homeassistant.api.check_api_state():
supervisor/api/proxy.py:        if not await self.sys_homeassistant.api.check_api_state():
supervisor/api/proxy.py:        if not await self.sys_homeassistant.api.check_api_state():
supervisor/auth.py:        if not await self.sys_homeassistant.api.check_api_state():
supervisor/backups/manager.py:            if not await self.sys_homeassistant.api.check_api_state():
supervisor/discovery/__init__.py:        if not await self.sys_homeassistant.api.check_api_state():
supervisor/homeassistant/api.py:    async def check_api_state(self) -> bool:
supervisor/homeassistant/core.py:            if await self.sys_homeassistant.api.check_api_state():
supervisor/homeassistant/websocket.py:        if not await self.sys_homeassistant.api.check_api_state():
supervisor/misc/tasks.py:        if await self.sys_homeassistant.api.check_api_state():
supervisor/resolution/notify.py:            or not await self.sys_homeassistant.api.check_api_state()
```

Reply:
```
        with suppress(HomeAssistantAPIError):
            data = await self.get_config()
            # Older versions of home assistant does not expose the state
            if data and data.get("state", "RUNNING") == "RUNNING":
                return True
        return False

```

So thats why its being called so frequently

Reply:
I think it would make sense to close this and add a new api to get the running state

Reply:
>  I wonder if its not working right and falling back because of our StrEnum backport.

If it is, that would be solved soon™️ once we drop 3.10 support.

Reply:
https://github.com/home-assistant/core/pull/96860

I think its a better solution to simply send supervisor only what is actually needs and leave this all as-is